### PR TITLE
Revert the lexing of `c"…"` string literals

### PR DESCRIPTION
**Pull request by fmease**
Fixes \[after beta-backport\] #113235.
Further progress is tracked in #113333.

This PR *manually* reverts parts of #108801 (since a git-revert would've been too coarse-grained & messy)
and git-reverts #111647.

CC @fee1-dead (#108801) @klensy (#111647)
r? @compiler-errors

@rustbot label F-c_str_literals beta-nominated

### CODE REVIEW

**Comment by fmease** [comment]
> The job mingw-check failed!

What is going on?

File: tests/ui/rfcs/rfc-3348-c-string-literals/issue-113235.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Regression test for issue #113235.
+// check-pass
+// edition:2015
```
Comment (fmease) [inline tests/ui/rfcs/rfc-3348-c-string-literals/issue-113235.rs:3]: Fyi, I know that the default edition of UI tests is 2015 but I feel like making it explicit is more robust and telling.
I can spruce up this regression test with an additional comment if you'd like me to.

**Comment by apiraino** [comment]
Beta backport approved as per compiler team [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202023-07-06/near/372912922)

@rustbot label +beta-accepted

**Pull request merged**